### PR TITLE
TPROD-399 Removed Doctrine deprecations from Campaign executioners.

### PR DESCRIPTION
**Pull request by biozshock**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ x ]
| Automated tests included?              | [ x ] 

#### Description:
Removed Doctrine deprecations from Campaign executioners.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create Decision, Action and Condition in Campaigns, with and without immediate actions.
3. Check they all worked fine.

### CODE REVIEW

File: app/bundles/CampaignBundle/Executioner/ContactFinder/InactiveContactFinder.php [hunk 1]
```diff
@@ -107,8 +112,9 @@ public function getContactCount($campaignId, array $decisionEvents, ContactLimit
     /**
      * Clear Lead entities from memory.
      */
-    public function clear()
+    public function clear(): void
     {
-        $this->leadRepository->clear();
+        $this->manager->flush();
+        $this->manager->clear();
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Executioner/ContactFinder/InactiveContactFinder.php]: Same as https://github.com/mautic/mautic/pull/12192#discussion_r1162940864

Comment (biozshock) [reply app/bundles/CampaignBundle/Executioner/ContactFinder/InactiveContactFinder.php]: Please check now.

**Review by escopecz — APPROVED** [review]
These changes make sense. Thanks! 👍

**Review by mabumusa1 — APPROVED** [review]
I tested and campaigns works as expected

**Pull request merged**